edi_storage/models/edi_backend.py

```diff
@@ -76,3 +81,47 @@ def _component_sort_key(self, component_class):
         return (
             1 if getattr(component_class, "_storage_backend_type", False) else 0,
         ) + res
+
+    def _cron_check_storage_pending_input(self, **kw):
+        for backend in self.filtered(lambda b: b.storage_id):
+            backend._check_storage_pending_input(**kw)
+
+    def _check_storage_pending_input(self, **kw):
+        self.ensure_one()
+        if not self.storage_id or not self.input_dir_pending:
+            _logger.info(
+                "%s ignored. No storage and/or input directory specified.", self.name
+            )
+            return False
+
+        pending_files = self.storage_id.list_files(self.input_dir_pending)
+        exchange_type = self.env["edi.exchange.type"].search(
+            self._domain_exchange_type_storage_pending_input()
+        )
+        if not len(exchange_type) == 1:
+            _logger.info("%s ignored. More than one exchange type found.", self.name)
+            return False
+        for file_name in pending_files:
+            existing = self.env["edi.exchange.record"].search(
```

Comment:
if you don't need the record:
```suggestion
            existing = self.env["edi.exchange.record"].search_count(
```

Reply:
Done!